L42-metadata-flags.md

```diff
@@ -0,0 +1,140 @@
+After setting “Wait For Ready” option to the channel construction method, the channel will automatically enable this mechanism for all the outgoing RPC calls. NOT only the first one. This usage will be consistent with other language’s implementation.
+
+Also, by adding per RPC variable, the client can customize each RPC call to leverage the “Wait For Ready” mechanism. This can give Python a little advantage over other languages.
+
+### The 3-states switch of “Wait For Ready” in C-Core
+The implementation of “Wait For Ready” flags is not as simple as a one bit switch. It contains two bits one is `GRPC_INITIAL_METADATA_WAIT_FOR_READY` another one is `GRPC_INITIAL_METADATA_WAIT_FOR_READY_EXPLICITLY_SET`. Although two bits can represent four states, three of them are valid.
+
+Assume `flag` is the flag variable we are manipulating.
+
+If the developer wants to use default value, then `flag = flag & ~WAIT_FOR_READY & ~WAIT_FOR_READY_EXPLICITLY_SET` (both 0).
+
+If the developer doesn't want to "wait for ready", then `flag = flag & ~WAIT_FOR_READY & WAIT_FOR_READY_EXPLICITLY_SET` (0, 1).
+
+If the developer wants to "wait for ready", then `flag = flag & WAIT_FOR_READY & WAIT_FOR_READY_EXPLICITLY_SET` (1, 1).
+
+### Related Proposals: 
+
+N/A
+
+## Proposal
+
+* Add an optional `wait_for_ready` variable to `MultiCallable` classes initialization methods. Default `None`, accept `bool`.
+* Per RPC level `wait_for_ready` variable can override upper level.
+* Import initial metadata flags constants from `grpc_types.h` to `grpc.pxi`.
+* (Suggesting) Add an optional `wait_for_ready` variable to `Channel` class initialization method. Default `None`, accept `bool`.
+
+### DEMO snippets 
+
+```Python
+# Per RPC level
+stub = ...Stub(...)
+
+stub.initial_rpc_call(..., wait_for_ready=True)
+stub.important_transaction_1(..., wait_for_ready=True)
+stub.unimportant_transaction_2(...)
+stub.important_transaction_3(..., wait_for_ready=True)
+stub.unimportant_transaction_4(...)
+# The unimportant transactions can be status report, or health check, etc.
+```
+
+```Python
+# Channel level
+channel = grpc.insecure_channel(..., wait_for_ready=True)
+stub = ...Stub(channel)
+
+stub.important_transaction_1(...)
+stub.important_transaction_2(...)
+stub.unimportant_transaction_3(...) # This will wait as well
+stub.important_transaction_4(...)
+```
+
+## Rationale
+
+### Default `None`, accept `bool` value
+* The `None` state defined that this value haven't been touched
+* It might be useful in some scenarios that interceptors wants to read its value and make changes
+
+### Whether to add channel level variable or not
+* The channel level variable is convenient and Pythonic
+* But it may create extra burden for future API evolution
+* **Please** comment about whether you feel like this design works for you or not
+
+### Separate variable vs. Add an option
+* Separate variable provides easy-to-use interface; just add an option can be done quickly and remove easily if needed.
```

Comment:
You mean from user point of view. Once an option is added in gRPC, it cannot be removed without breaking backward compatibility.